policy-controller/src/main.rs

```diff
@@ -318,3 +257,104 @@ async fn load_private_key(filename: &str) -> anyhow::Result<rustls::PrivateKey>
 
     Ok(rustls::PrivateKey(keys[0].clone()))
 }
+
+/// Bind the specified address and serve the admission controller using the
+/// provided kubernetes client.
+async fn bind_admission_controller(
+    bind_addr: SocketAddr,
+    client: kube::Client,
+) -> Result<(
+    SocketAddr,
+    impl std::future::Future<Output = ()> + Send + 'static,
+)> {
+    let tcp = TcpListener::bind(&bind_addr).await?;
+    let listen_addr = tcp.local_addr()?;
+    // Connection accept loop
+    let serve = async move {
+        loop {
+            let socket = match tcp.accept().await {
+                Ok((socket, _)) => socket,
+                Err(err) => {
+                    error!(%err, "Failed to accept connection");
+                    continue;
+                }
+            };
+            let client_addr = match socket.peer_addr() {
+                Ok(addr) => addr,
+                Err(error) => {
+                    error!(%error, "Failed to get peer address");
+                    continue;
+                }
+            };
+
+            tokio::spawn(
+                serve_admission_controller_conn(socket, client.clone())
+                    .instrument(info_span!("connection", client.addr = %client_addr)),
+            );
+        }
```

Comment:
here we spawn a task as quickly as possible, though we first read the client's address so we can put it on the tracing context